apiserver/facades/client/application/deployrepository.go

```diff
@@ -619,9 +620,7 @@ func (v *deployFromRepositoryValidator) deducePlatform(arg params.DeployFromRepo
 			return corecharm.Platform{}, usedModelDefaultBase, err
 		}
 		platform.OS = base.OS
-		// platform channels don't model the concept of a risk
-		// so ensure that only the track is included
-		platform.Channel = base.Channel.Track
+		platform.Channel = base.Channel.String()
```

Comment:
Explain this change?

Reply:
Charmhub doesn't compute the risk of a juju base correctly and will error. We had been accounting for this in many places, but recently moved the clean up into the charm hub code. It's no longer needed in other places.